src/modules/forms/EndpointForm/EndpointForm.js

```diff
@@ -57,415 +57,432 @@ const propTypes = {
 }
 
 class EndpointForm extends PureComponent {
-    state = {
-      upstreams: this.props.initialValues.proxy.upstreams || {} // fallback for old endpoints (they have `upstreams: null`), probably temporary
-    };
+  state = {
+    showJSONmodal: false,
+    JSONmodalContent: {},
+    upstreams: this.props.initialValues.proxy.upstreams || {} // fallback for old endpoints (they have `upstreams: null`), probably temporary
```

Comment:
hm it should not be coming with `null` value I guess, do you remember which proxy does?